### PR TITLE
xchain-aggregator PoC

### LINKED ISSUE #403: xchainjs-aggregator
## Description
begin implementing aggregator

### PR DESCRIPTION
**Pull request by cinnamoroll6130**

### CODE REVIEW

File: src/renderer/components/swap/Swap.tsx [hunk 1]
```diff
@@ -1291,91 +1063,112 @@ export const Swap = ({
...
+        if (!isTokenAsset(sourceAsset) && !isTradeAsset(sourceAsset) && !isSynthAsset(sourceAsset)) {
+          if (sourceChainAssetAmount.lt(amountToSwap.plus(swapFees.inFee.amount))) {
+            amountToSwap = sourceChainAssetAmount.minus(swapFees.inFee.amount)
+          }
+        }
+        return {
+          poolAddress,
+          asset: sourceAsset,
+          amount: amountToSwap,
+          memo: updateMemo(quoteSwap.memo, applyBps, network),
```
Comment (Thorian1te) [inline src/renderer/components/swap/Swap.tsx:962]: @cinnamoroll6130 
 I update the memo here based applyBps

File: src/renderer/views/swap/SwapView.tsx [hunk 2]
```diff
@@ -138,49 +128,96 @@ const SuccessRouteView: React.FC<Props> = ({
     keystoreService: { keystoreState$, validatePassword$ }
   } = useWalletContext()
 
+  const { chain: sourceChain } = sourceAsset.type === AssetType.SYNTH ? mayaDetails.asset : sourceAsset
+  const { chain: targetChain } = targetAsset.type === AssetType.SYNTH ? mayaDetails.asset : targetAsset
+
+  const poolAddressThor$ = poolAddressesByChain$(sourceChain)
+  const poolAddressMaya$ = poolAddressesByChainMaya$(sourceChain)
+  const combinedPoolAddresses = useObservableState(
```
Comment (cinnamoroll6130) [inline src/renderer/views/swap/SwapView.tsx]: @Thorian1te, Please double-check the logic here, this is causing an infinite re-render

File: src/renderer/views/swap/SwapView.tsx [hunk 3]
```diff
@@ -138,49 +128,96 @@ const SuccessRouteView: React.FC<Props> = ({
...
+    FP.pipe(
+      Rx.combineLatest([poolAddressThor$, poolAddressMaya$]).pipe(
+        RxOp.map(([thorPoolAddressRD, mayaPoolAddressRD]) =>
+          FP.pipe([thorPoolAddressRD, mayaPoolAddressRD], A.map(RD.toOption), ([thorOption, mayaOption]) => ({
+            thor: thorOption, // Include the ThorChain pool address
+            maya: mayaOption // Include the MayaChain pool address
+          }))
+        ),
+        RxOp.distinctUntilChanged(), // Avoid emitting the same values multiple times
+        RxOp.debounceTime(100) // Throttle frequent updates (adjust time as needed)
```
Comment (cinnamoroll6130) [inline src/renderer/views/swap/SwapView.tsx]: It's re-renders in every 100 ms. @Thorian1te

File: src/renderer/views/swap/SwapView.tsx (hunk 2 above)
Comment (Thorian1te) [reply src/renderer/views/swap/SwapView.tsx]: ok

File: src/renderer/views/swap/SwapView.tsx (hunk 3 above)
Comment (Thorian1te) [reply src/renderer/views/swap/SwapView.tsx]: ok

**Review by Thorian1te — APPROVED** [review]
lgtm

**Comment by gdoteof** [comment]
fwiw, i am pretty sure this broke bitcoin swaps.  i have only had the code checked out for a few hours but after trying to do a bitcoin->rune swap, i eventually checked out 1.22.9 and it was working.

it looks to me like the app can't find a 'router' for bitcoin, not sure what that is but presumably the place to broadcast transactions

**Pull request merged**